tests/chainer_tests/functions_tests/rnn_tests/test_function_n_step_rnn.py

```diff
@@ -105,261 +137,130 @@ def check_forward(
-        self.ws = []
-        self.bs = []
-        # First layer has the different shape
-        for di in range(2):
-            self.ws.append(_shaped_random([(o, i), (o, o)]))
-            self.bs.append(_shaped_random([o, o]))
-        # Rest layers
-        for _ in range(self.n_layers - 1):
-            for di in range(2):
-                self.ws.append(_shaped_random([(o, o * 2), (o, o)]))
-                self.bs.append(_shaped_random([o, o]))
-
-        self.dys = _shaped_random(
-            [(b, self.out_size * 2) for b in self.batches])
-        self.dhy = _shaped_random(h_shape)
-
-    def check_forward(
-            self, h_data, xs_data, ws_data, bs_data):
-        h = _wrap_variable(h_data)
-        xs = _wrap_variable(xs_data)
-        ws = _wrap_variable(ws_data)
-        bs = _wrap_variable(bs_data)
-        hy, ys = functions.n_step_birnn(
-            self.n_layers, self.dropout, h, ws, bs, xs,
-            activation=self.activation)
-
-        xs_next = self.xs
-        e_hy = self.hx.copy()
+        self.check_forward_options = {'atol': 1e-3, 'rtol': 1e-2}
+        self.check_backward_options = {'atol': 5e-3, 'rtol': 5e-2}
+        self.check_double_backward_options = {'atol': 5e-3, 'rtol': 5e-2}
+        self.skip_double_backward_test = True
+
+    def generate_inputs(self):
+        h_shape = (self.n_layers * 2, self.batches[0], self.hidden_size)
+        dtype = numpy.float32
+
+        h = array(h_shape, dtype)
+        in_size = self.input_size
+        out_size = self.hidden_size
+        xs = [array((self.batches[b], in_size), dtype)
+              for b in range(len(self.batches))]
+
+        def w_in(i, j):
+            if i == 0 and j < 1:
+                return in_size
+            elif i > 0 and j < 1:
+                return out_size * 2
+            else:
+                return out_size
+
+        inputs = []
+        inputs.append(h)
+        for i in range(len(self.batches)):
+            inputs.append(xs[i])
+
+        for n in range(self.n_layers):
+            for direction in (0, 1):
+                for i in range(2):
+                    inputs.append(array((out_size, w_in(n, i)), dtype))
+                for i in range(2):
+                    inputs.append(array((out_size,), dtype))
+        return tuple(inputs)
+
+    def process_inputs(self, inputs):
+        h = inputs[0]
+        xs = inputs[1: 1 + len(self.batches)]
+        ws = []
+        bs = []
+        index = 1 + len(self.batches)
+        for n in range(self.n_layers):
+            ws.append(inputs[index: index + 2])
+            bs.append(inputs[index + 2: index + 4])
+            ws.append(inputs[index + 4: index + 6])
+            bs.append(inputs[index + 6: index + 8])
+            index += 8
+        return h, ws, bs, xs
+
+    def forward(self, inputs, device):
+        h, ws, bs, xs = self.process_inputs(inputs)
+
+        if h.array.dtype == numpy.float64:
+            with chainer.using_config('use_cudnn', 'never'):
```

Comment:
Now with `numerical_grad_dtype = None`, I think this workaround can be removed too,

Reply:
Done